### PR TITLE
Demote/reject teacher assistants

### PR DESCRIPTION
**Pull request by 0xf8f8ff**
- course teachers can now be demoted to students or rejected from course by the course creator
- course creator cannot be demoted/rejected

### CODE REVIEW

File: web/access_control.go [hunk 1]
```diff
@@ -132,6 +132,25 @@ func (s *AutograderService) isTeacher(userID, courseID uint64) bool {
...
+
+	if (currEnrol.GetStatus() == pb.Enrollment_TEACHER) && (enrol.GetStatus() != pb.Enrollment_TEACHER) {
+		return true
+	}
+	return false
+}
+
+// isCourseCreator returns true if the given user is course creator for the given course
+func (s *AutograderService) isCourseCreator(courseID, userID uint64) bool {
+	course, _ := s.db.GetCourse(courseID, false)
```
Comment (meling) [inline web/access_control.go]: GetCourse can return `nil`; so here you should also check for `err != nil` and if so, just return false.

Comment (meling) [reply web/access_control.go]: Or alternatively, check that `course != nil` before using it for `GetCourseCreatorID()`.

Comment (0xf8f8ff) [reply web/access_control.go]: But if `course` is nil, `course.GetCourseCreatorID()` should return 0, shouldn't it?

Comment (meling) [reply web/access_control.go]: That is correct. So this should work fine.

**Pull request merged**